### ISSUE TITLE
Uberdog Server Issue

### ISSUE THREAD

**Issue by mnmarinaccio** [open-toontown/open-toontown#26]
When trying to start the uberdog server for win32, the console is saying it is not finding the pytz module in the ToontownTimeManager.py file. I'm not too familiar with python, I only know C, but I looked inside that file to try and troubleshoot. Is pytz just a module that is not used anymore, but the "import pytz" line of code was not commented out, or is there maybe a file I am missing on my computer that didn't download when I downloaded all of the files? Any help is appreciated, thank you!

**Comment by rocketprogrammer** [comment]
pytz is a dependency. You need to install it using pip.
Say my panda is installed at C:/Panda3D-1.11.0-x64-py3.9, I would do C:/Panda3D-1.11.0-x64-py3.9/python/python.exe -m pip install pytz.

**Comment by mnmarinaccio** [comment]
Oh okay perfect, thank you.

**Comment by danr789** [comment]
If you are using Windows, then pytz will not work (for some mysterious reasons), I suggest delete ToontownTimeManager.py and replace that with the 2 Toontown Fellowship Code for Time Manager, it will work fine.

File 1 : https://github.com/toontownfellowship/src/blob/master/toontown/parties/ToontownTimeManager.py

File 2:  https://github.com/toontownfellowship/src/blob/master/toontown/parties/ToontownTimeZone.py

Please Note : Both Files are Needed, Replace the old file with the 2 new ones in the same location.

**Comment by mnmarinaccio** [comment]
Cool, thank you very much. I am very new to python so I'm trying to figure out all of it's syntax and stuff. I will try the two files provided, thank you.

**Comment by rocketprogrammer** [comment]
Pytz... works on windows?
What are you talking about

**Comment by padlocks** [comment]
My best guess is you tried to install pytz on the wrong Python installation rather the one bundled with Panda3D. Pytz installed and the game is running as expected. The command stated above directly asks the bundled Python to install pytz.

PPYTHON_PATH in the base directory tells the game which Python installation to use, but any other installation other than the Panda3D bundle won't work properly. If you install the bundle in a non-default location, this would be the way to connect the two.

**Comment by danr789** [comment]
> My best guess is you tried to install pytz on the wrong Python installation rather the one bundled with Panda3D. Pytz installed and the game is running as expected. The command stated above directly asks the bundled Python to install pytz.
> 
> PPYTHON_PATH in the base directory tells the game which Python installation to use, but any other installation other than the Panda3D bundle won't work properly. If you install the bundle in a non-default location, this would be the way to connect the two.

You are right ( in a way ) , the installed version of python will gladly download pytz, BUT the one that came with panda3D will give out a error saying file not found.

**Comment by danr789** [comment]
> If you are using Windows, then pytz will not work (for some mysterious reasons), I suggest delete ToontownTimeManager.py and replace that with the 2 Toontown Fellowship Code for Time Manager, it will work fine.
> 
> File 1 : https://github.com/toontownfellowship/src/blob/master/toontown/parties/ToontownTimeManager.py
> 
> File 2: https://github.com/toontownfellowship/src/blob/master/toontown/parties/ToontownTimeZone.py
> 
> Please Note : Both Files are Needed, Replace the old file with the 2 new ones in the same location.

But this worked fine for me.

**Comment by Vitor23laser** [comment]
guy the rocket is right

**Comment by Vitor23laser** [comment]
If you have Windows 11, go to the folder C:\Panda3D-1.10.16-x64\python, right-click on the mouse, go to the terminal and write pip install pytz.
